### ISSUE TITLE
flow vs fair queueing

### ISSUE THREAD

**Issue by dtaht** [muxamilian/fq-detection-fast#1]
I am pleased to see this in C. And you are a doctor now! Congrats both ways.

I make a distinction between fair and flow queuing. Fair queuing is usually round robin, flow queuing is if the arrival rate of a flow is less than the departure rate of all the other flows, it goes out first.

Being able to detect the difference between fair and flow queuing should be straightforward.

**Comment by muxamilian** [comment]
Thanks for your suggestions! 

So you mean "flow queuing" is like "sparse flow optimization" in fq_codel (https://ieeexplore.ieee.org/document/8469111)?

In recent papers it seems to me that people use "fair queuing" and "flow queuing" as synonyms (for example https://arxiv.org/pdf/2107.00904.pdf).

**Comment by dtaht** [comment]
yes, an error I have been trying to correct for years. Even well informed people keep making the strict round robin assumption for it, and are grievously wrong...  and I am a contributor to it by never explaining when I say fair(flow) queuing or "fair/flow queuing" in a sentence that they are not equivalent. Been meaning to try and write that up for years and years... Anyway, this shows a clear difference between sfq and flow queuing, if you log in.

https://forum.mikrotik.com/viewtopic.php?t=179307#p885613

**Comment by dtaht** [comment]
also thx for the steer to the paper. flow queuing and videoconferencing are kind of an unexplored area. IF more videoconferencing systems did packet pacing (none that I know of do), flow queuing would show some sort of difference.

I am also newly aware that pion´s gcc implementation changes rate based on loss statistics only, not delay. I *think* the fq_codel result for zoom shows they are loss based.
